test/integration/Reputation.spec.ts

```diff
@@ -0,0 +1,108 @@
+
+const Reputation = require('@daostack/arc/build/contracts/Reputation.json');
+
+describe('Reputation', () => {
+  let web3, addresses, reputation;
+  beforeAll(async () => {
+    web3 = await getWeb3();
+    addresses = getContractAddresses();
+    const opts = await getOptions(web3);
+    reputation = new web3.eth.Contract(Reputation.abi, addresses.Reputation, opts);
+  });
+
+  it('Sanity', async () => {
+    const accounts = web3.eth.accounts.wallet;
+    let txs = [];
+    txs.push(await reputation.methods.mint(accounts[0].address, '100').send());
+    txs.push(await reputation.methods.mint(accounts[1].address, '100').send());
+    txs.push(await reputation.methods.burn(accounts[0].address, '30').send());
+    txs.push(await reputation.methods.mint(accounts[2].address, '300').send());
+    txs.push(await reputation.methods.burn(accounts[1].address, '100').send());
+    txs.push(await reputation.methods.burn(accounts[2].address, '1').send());
+    txs = txs.map(({ transactionHash }) => transactionHash);
+
+    const { reputationHolders } = await query(`{
+      reputationHolders {
+        contract,
+        address,
+        balance
+      }
+    }`);
+
+    expect(reputationHolders.length).toEqual(2);
+    expect(reputationHolders).toContainEqual({
+      contract: reputation.options.address.toLowerCase(),
+      address: accounts[0].address.toLowerCase(),
+      balance: '70'
+    })
+    expect(reputationHolders).not.toContainEqual({
+      contract: reputation.options.address.toLowerCase(),
+      address: accounts[1].address.toLowerCase(),
+      balance: '0'
+    })
+    expect(reputationHolders).toContainEqual({
+      contract: reputation.options.address.toLowerCase(),
+      address: accounts[2].address.toLowerCase(),
+      balance: '299'
+    })
+
+    const { reputationMints } = await query(`{
+      reputationMints {
+        txHash,
+        contract,
+        address,
+        amount
+      }
+    }`);
+
+    expect(reputationMints.length).toEqual(3);
+    expect(reputationMints).toContainEqual({
+      txHash: txs[0],
+      contract: reputation.options.address.toLowerCase(),
+      address: accounts[0].address.toLowerCase(),
+      amount: '100'
+    });
+    expect(reputationMints).toContainEqual({
+      txHash: txs[1],
+      contract: reputation.options.address.toLowerCase(),
+      address: accounts[1].address.toLowerCase(),
+      amount: '100'
+    });
+    expect(reputationMints).toContainEqual({
+      txHash: txs[3],
+      contract: reputation.options.address.toLowerCase(),
+      address: accounts[2].address.toLowerCase(),
+      amount: '300'
+    });
+
+    const { reputationBurns } = await query(`{
+      reputationBurns {
+        txHash,
+        contract,
+        address,
+        amount
+      }
+    }`);
+
+    expect(reputationBurns.length).toEqual(3);
+    expect(reputationBurns).toContainEqual({
+      txHash: txs[2],
+      contract: reputation.options.address.toLowerCase(),
```

Comment:
can we set the address type to be lower case in the db by default ?

Reply:
Its already lowercase in the db. Its mixed case in web3.

Reply:
addresses are saved in lowercase in the db, eg. `0xa1b...`. web3 returns addresses in mixed case, eg. `0xa1B...`. we need to convert the web3 address to lowercase in order to match it to the one the api returns (which is lowercase).